Gems/ScriptCanvas/Code/CMakeLists.txt

```diff
@@ -89,6 +89,22 @@ ly_add_target(
         Gem::ScriptCanvasDebugger
 )
 
+ly_add_target(
+    NAME ScriptCanvasAPI HEADERONLY
```

Comment:
To follow the convention of other targets, I would name this "ScriptCanvas.API` instead. Since this is going to become a common pattern, we may want a more explicit/standardized name for it, @lawsonamzn may have better suggestions for the name.

Reply:
I'm making a similar change for LmbrCentral to address a similar issue.  I'll update this in that PR.